libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -154,3 +176,13 @@ bool SyntaxChecker::visit(PlaceholderStatement const&)
 	return true;
 }
 
+bool SyntaxChecker::visit(MemberAccess const& _memberAccess)
+{
+	if (m_nonPayablePublic && !m_library) {
+		auto const& identifier = dynamic_cast<Identifier const&>(_memberAccess.expression());
```

Comment:
You are right. Sounds like an interesting project to take. I'll see what I can do and get back to you when I have a prototype for the StaticAnalyzer.